Add repositoryId overloads to methods on I(Observable)RepoCollaboratorsClient

dampir opened the pull request:
As part of my work on #1120 I've added new overloads on I(Observable)RepoCollaboratorsClient to get access by repository id.

- [x] **Rename parameter "repo" to "name" (in order to deliver more consistency into the codebase).**

	 I've found out that there is the pair (owner, name) is used to identify repository. But some clients use pair (owner, repo) and it's some type of inconsistency.
	  So I've decided to fix these cases during my work on #1120.
- [x] **Update XML documentation of interface methods of clients (also synchronize XML docs of IRepoCollaboratorsClient and IObservableRepoCollaboratorsClient).**

	  There is some divergence between XML documentation of methods in IRepoCollaboratorsClient and IObservableRepoCollaboratorsClient. So I've decided 
	  to sync XML documentation of these classes during my work on #1120.
- [x] **Add overloads to IRepoCollaboratorsClient.**

	  Just add overloads of existing methods that use repositoryId to work with repo.
- [x] **Add overloads to IObservableRepoCollaboratorsClient.**

	  Just add overloads of existing methods that use repositoryId to work with repo.
- [x] **Add unit tests.**

	  I've added new unit tests that use repositoryId to work with repo that is just a full copy of existing tests that use (owner, name) key.
	  Also I've found out that not all methods are covered by tests and added them for new and for old methods.
- [x] **Add integration tests.**

	  I've added new integration tests that use repositoryId to work with repo that is just a full copy of existing tests that use (owner, name) key.
	  Also I've found out that not all methods are covered by tests and added them for new and for old methods.

I think that we should discuss and review this PR as thorough as we can because it's first PR in series of PRs that should add ability to work with repo by their Id. :smile: 

/cc @shiftkey, @ryangribble

ryangribble commented:
I'll have a proper thorough look at this on the weekend 

But I just wanted to say that I'm wondering from a high level release management perspective, if it might be a good approach to get the ApiOptions pagination work finished first, then cut a new octokit release (as there a number of fixes/changes that would be good to get released), prior to embarking on the large changes to implement the repositoryID overloads.  I'm definitely not saying you shouldn't work on these but I'd be keen to get a release happening soon. Perhaps we don't need all ApiOptions done and perhaps it doesn't matter if some repositoryID changes are in that release either, although it also does seem clean to think that the next release would have all the ApiOptions pagination in it, then the next release all the repositoryID changes

dampir commented:
@ryangribble yes, you're definetely right! I'm also think that "ApiOption" release should be deliver before "repositoryId" one.
And maybe it is would be helphul, here is the list of clients that should get "repositoryId" overload through GSoC 2016 and I've marked items that have already ApiOptions overloads, 
added issue link with [WIP] prefix if issue has been grabbed by someone 
and just added link to open issue if there is nobody is working on the issue:

- [x] Collaborators #1170 (IRepoCollaboratorsClient, IObservableRepoCollaboratorsClient)
- [ ] Comments [WIP] #1262 (IRepositoryCommentsClient, IObservableRepositoryCommentsClient)
- [x] Commits #1173 (IRepositoryCommitsClient, IObservableRepositoryCommitsClient)
- [ ] Contents #1174 (IRepositoryContentsClient, IObservableRepositoryContentsClient)
- [ ] Deploy Keys #1175 (IRepositoryDeployKeysClient, IObservableRepositoryDeployKeysClient)
- [x] Deployments #1151 (IDeploymentsClient, IObservableDeploymentsClient)
- [x] Forks #1176 (IRepositoryForksClient, IObservableRepositoryForksClient)
- [x] ~~Merging (IMergingClient, IObservableMergingClient)~~ (no need to add ApiOptions overloads)
- [ ] Pages [WIP] #1304 (IRepositoryPagesClient, IObservableRepositoryPagesClient)
- [ ] Releases #1169 (IReleasesClient, IObservableReleasesClient)
- [ ] Statistics (IStatisticsClient, IObservableStatisticsClient)
	  
	  There is no issue, but I think that it should be added because of IStatisticsClient.GetContributors(string owner, string repositoryName) method?
	  Am I right?
- [x] Statuses #1150 (ICommitStatusClient, IObservableCommitStatusClient)
- [x] Webhooks #1177 (IRepositoryHooksClient, IObservableRepositoryHooksClient)

I think that I could do these issues with ApiOptions before work on repositoryId overloads, as far as I can.
